build: Update Storybook version

mfarukkoc opened the pull request:
### Description
- Storybook version is updated to `6.3.2` from `5.3.19`.

### Notes
- You may need to remove and reinstall the `node_modules` folder.

Closes #89

mucahit reviewed (approved):
Congrats on your first PR :tada:

yasincaliskan commented:
I just wonder, should we update config files? @mucahit 
Please take a look [here](https://storybook.js.org/blog/storybook-6-migration-guide/)

mucahit commented:
> I just wonder, should we update config files? @mucahit
> Please take a look [here](https://storybook.js.org/blog/storybook-6-migration-guide/)

Thanks for that @yasincaliskan It seems we don't need to update our configs.

edizcelik commented:
I got this error while using Node `v14.15.4` and NPM `6.14.10`:

```
➜ npm run storybook

> @hipo/react-ui-toolkit@1.0.0-alpha.3.2.0 storybook /Users/edizcelik/Documents/Hipo/react-ui-toolkit
> start-storybook -p 6006

info @storybook/react v6.3.2
info
info => Loading presets
WARN   Failed to load preset: "/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/manager-webpack4/dist/cjs/presets/manager-preset.js"
ERR! Error: Cannot find module 'react/package.json'
ERR! Require stack:
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/ui/noop.js
ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
ERR!     at resolveFileName (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/resolve-from/index.js:29:39)
ERR!     at resolveFrom (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/resolve-from/index.js:43:9)
ERR!     at module.exports (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/resolve-from/index.js:46:47)
ERR!     at Object.<anonymous> (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/ui/paths.js:17:18)
ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
ERR!     at Module.require (internal/modules/cjs/loader.js:952:19)
ERR!     at require (internal/modules/cjs/helpers.js:88:18)
ERR!     at Object.<anonymous> (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/manager-webpack4/dist/cjs/manager-webpack.config.js:50:38)
ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
ERR!  Error: Cannot find module 'react/package.json'
ERR! Require stack:
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/ui/noop.js
ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
ERR!     at resolveFileName (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/resolve-from/index.js:29:39)
ERR!     at resolveFrom (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/resolve-from/index.js:43:9)
ERR!     at module.exports (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/resolve-from/index.js:46:47)
ERR!     at Object.<anonymous> (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/ui/paths.js:17:18)
ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
ERR!     at Module.require (internal/modules/cjs/loader.js:952:19)
ERR!     at require (internal/modules/cjs/helpers.js:88:18)
ERR!     at Object.<anonymous> (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/manager-webpack4/dist/cjs/manager-webpack.config.js:50:38)
ERR!     at Module._compile (internal/modules/cjs/loader.js:1063:30)
ERR!     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
ERR!     at Module.load (internal/modules/cjs/loader.js:928:32)
ERR!     at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
ERR!   code: 'MODULE_NOT_FOUND',
ERR!   requireStack: [
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/ui/noop.js'
ERR!   ]
ERR! }
(node:32111) DeprecationWarning: Configuration files such as "config", "presets" and "addons" are deprecated and will be removed in Storybook 7.0.
Read more about it in the migration guide: https://github.com/storybookjs/storybook/blob/master/MIGRATION.md#to-mainjs-configuration
(Use `node --trace-deprecation ...` to show where the warning was created)
info => Loading 1 config file in "/Users/edizcelik/Documents/Hipo/react-ui-toolkit/.storybook"
info => Loading custom manager config
ERR! Error: Cannot find module 'react/package.json'
ERR! Require stack:
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/preview/iframe-webpack.config.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/presets/preview-preset.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-common/dist/cjs/presets.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-common/dist/cjs/index.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core/dist/cjs/server.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core/server.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/react/dist/cjs/server/index.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/react/bin/index.js
ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
ERR!     at Function.resolve (internal/modules/cjs/helpers.js:94:19)
ERR!     at _default (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/preview/iframe-webpack.config.js:190:46)
ERR!     at async Object.start (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/index.js:95:16)
ERR!     at async Promise.all (index 0)
ERR!     at async storybookDevServer (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/dev-server.js:123:28)
ERR!     at async buildDevStandalone (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/build-dev.js:112:31)
ERR!     at async buildDev (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/build-dev.js:154:5)
ERR!  Error: Cannot find module 'react/package.json'
ERR! Require stack:
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/preview/iframe-webpack.config.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/presets/preview-preset.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-common/dist/cjs/presets.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-common/dist/cjs/index.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core/dist/cjs/server.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core/server.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/react/dist/cjs/server/index.js
ERR! - /Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/react/bin/index.js
ERR!     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
ERR!     at Function.resolve (internal/modules/cjs/helpers.js:94:19)
ERR!     at _default (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/preview/iframe-webpack.config.js:190:46)
ERR!     at async Object.start (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/index.js:95:16)
ERR!     at async Promise.all (index 0)
ERR!     at async storybookDevServer (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/dev-server.js:123:28)
ERR!     at async buildDevStandalone (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/build-dev.js:112:31)
ERR!     at async buildDev (/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/build-dev.js:154:5) {
ERR!   code: 'MODULE_NOT_FOUND',
ERR!   requireStack: [
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/preview/iframe-webpack.config.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/builder-webpack4/dist/cjs/presets/preview-preset.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-common/dist/cjs/presets.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-common/dist/cjs/index.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core-server/dist/cjs/index.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core/dist/cjs/server.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/core/server.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/react/dist/cjs/server/index.js',
ERR!     '/Users/edizcelik/Documents/Hipo/react-ui-toolkit/node_modules/@storybook/react/bin/index.js'
ERR!   ]
ERR! }

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @hipo/react-ui-toolkit@1.0.0-alpha.3.2.0 storybook: `start-storybook -p 6006`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @hipo/react-ui-toolkit@1.0.0-alpha.3.2.0 storybook script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/edizcelik/.npm/_logs/2021-07-09T09_57_18_575Z-debug.log
```

`npm run dev` works without any issues but `storybook` script has failed with the error above. I removed node_modules and reinstalled the whole thing but still got the same error.

Please also notice the following warning:

```
(node:32111) DeprecationWarning: Configuration files such as "config", "presets" and "addons" are deprecated and will be removed in Storybook 7.0.
Read more about it in the migration guide: https://github.com/storybookjs/storybook/blob/master/MIGRATION.md#to-mainjs-configuration
(Use `node --trace-deprecation ...` to show where the warning was created)
```

Then, I upgraded my NPM version to `7.19.1`, using still the same Node v14.17.3, I reinstalled the whole thing and now it works. But we still have these deprecation warning on the console:

```
info => Loading presets
(node:39803) DeprecationWarning: Configuration files such as "config", "presets" and "addons" are deprecated and will be removed in Storybook 7.0.
Read more about it in the migration guide: https://github.com/storybookjs/storybook/blob/master/MIGRATION.md#to-mainjs-configuration
(Use `node --trace-deprecation ...` to show where the warning was created)
```

```
info => Loading custom Webpack config (full-control mode).
(node:39803) DeprecationWarning: Default PostCSS plugins are deprecated. When switching to '@storybook/addon-postcss',
you will need to add your own plugins, such as 'postcss-flexbugs-fixes' and 'autoprefixer'.

See https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated-default-postcss-plugins for details.
```

mucahit commented:
> I got this error while using Node `v14.15.4` and NPM `6.14.10`:
> 

Thanks, @edizcelik. It's interesting I haven't got those errors, let's take a detailed look at this @mfarukkoc

mfarukkoc commented:
@edizcelik 
I think the first error is because the peer dependencies are not installed. I get npm warnings below when I run the `npm install`  on Node `v14.15.4` and NPM `6.14.10`:
```
...
npm WARN @hipo/react-ui-toolkit@1.0.0-alpha.3.2.0 requires a peer of react@17.0.2 but none is installed. You must install peer dependencies yourself.
npm WARN @hipo/react-ui-toolkit@1.0.0-alpha.3.2.0 requires a peer of react-dom@17.0.2 but none is installed. You must install peer dependencies yourself.
npm WARN @storybook/addons@6.3.2 requires a peer of react@^16.8.0 || ^17.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @storybook/addons@6.3.2 requires a peer of react-dom@^16.8.0 || ^17.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @storybook/react@6.3.2 requires a peer of react@^16.8.0 || ^17.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @storybook/react@6.3.2 requires a peer of react-dom@^16.8.0 || ^17.0.0 but none is installed. You must install peer dependencies yourself.
...
```
I fixed this problem by adding `react` and `react-dom` peer dependencies to the dev dependency.

I updated the config files to remove the deprecation warnings.

mucahit reviewed (approved):
Works fine with node@v14.15.4 and npm@7.12.1
